### PR TITLE
MAYA-102108 - merge ae work back to refactoring sandbox

### PR DESCRIPTION
**Pull request by seando-adsk**
Merged github/dev into github/refactoring_sandbox

### CODE REVIEW

File: doc/DEVELOPER.md [hunk 1]
```diff
@@ -12,7 +12,7 @@
 
 |                     | Pixar USD         | AL_USDMaya          |
 | ------------------- | ----------------- | ------------------- |
-| commit/tag          | fe3b149  (>19.11) | 631a2911 (>0.34.0)  |
+| commit/tag          | 2ef943b  (>19.11) | 631a2911 (>0.34.0)  |
```
Comment (seando-adsk) [inline doc/DEVELOPER.md]: Not my change, but this is the right commit

File: lib/render/vp2RenderDelegate/mesh.cpp [hunk 2]
```diff
@@ -99,6 +99,9 @@ namespace {
         //! Color array to support per-instance color and selection highlight.
         MFloatArray _instanceColors;
 
+        //! If true, associate geometric buffers to the render item and trigger consolidation/instancing update
+        bool _geometryDirty{ false };
```
Comment (seando-adsk) [inline lib/render/vp2RenderDelegate/mesh.cpp]: These aren't my changes, are the correct?

File: lib/render/vp2ShaderFragments/shaderFragments.cpp [hunk 3]
```diff
@@ -110,15 +110,19 @@ namespace
...
 {
-    MHWRender::MRenderer* theRenderer = MHWRender::MRenderer::theRenderer();
+    // We do not force the renderer to initialize in case we're running in a
+    // headless context. If we cannot get a handle to the renderer or the
+    // fragment manager, we assume that's the case and simply return success.
+    MHWRender::MRenderer* theRenderer =
+        MHWRender::MRenderer::theRenderer(/* initializeRenderer = */ false);
     if (!theRenderer) {
-        return MS::kFailure;
+        return MS::kSuccess;
```
Comment (seando-adsk) [inline lib/render/vp2ShaderFragments/shaderFragments.cpp]: These aren't my changes, are the correct?

File: plugin/pxr/maya/plugin/pxrUsdPreviewSurface/plugin.cpp [hunk 4]
```diff
@@ -78,15 +78,19 @@ static
...
 {
-    MHWRender::MRenderer* theRenderer = MHWRender::MRenderer::theRenderer();
+    // We do not force the renderer to initialize in case we're running in a
+    // headless context. If we cannot get a handle to the renderer or the
+    // fragment manager, we assume that's the case and simply return success.
+    MHWRender::MRenderer* theRenderer =
+        MHWRender::MRenderer::theRenderer(/* initializeRenderer = */ false);
     if (!theRenderer) {
-        return MS::kFailure;
+        return MS::kSuccess;
```
Comment (seando-adsk) [inline plugin/pxr/maya/plugin/pxrUsdPreviewSurface/plugin.cpp]: These aren't my changes, are the correct?

File: doc/DEVELOPER.md (hunk 1 above)
Comment (kxl-adsk) [reply doc/DEVELOPER.md]: Yes, comes from #133

File: lib/render/vp2RenderDelegate/mesh.cpp (hunk 2 above)
Comment (kxl-adsk) [reply lib/render/vp2RenderDelegate/mesh.cpp]: Yes, come from #139

File: lib/render/vp2ShaderFragments/shaderFragments.cpp (hunk 3 above)
Comment (kxl-adsk) [reply lib/render/vp2ShaderFragments/shaderFragments.cpp]: Yes, #130

File: plugin/pxr/maya/plugin/pxrUsdPreviewSurface/plugin.cpp (hunk 4 above)
Comment (kxl-adsk) [reply plugin/pxr/maya/plugin/pxrUsdPreviewSurface/plugin.cpp]: Yes, #130

**Pull request merged**